libbpf-tools/mountsnoop.bpf.c

```diff
@@ -57,16 +46,17 @@ static int probe_exit(void *ctx, int ret)
 	__u64 pid_tgid = bpf_get_current_pid_tgid();
 	__u32 pid = pid_tgid >> 32;
 	__u32 tid = (__u32)pid_tgid;
-	struct arg *argp;
-	struct event *eventp;
 	struct task_struct *task;
-	int zero = 0;
+	struct event *eventp;
+	struct arg *argp;
 
 	argp = bpf_map_lookup_elem(&args, &tid);
 	if (!argp)
 		return 0;
 
-	eventp = bpf_map_lookup_elem(&heap, &zero);
+	bpf_map_delete_elem(&args, &tid);
```

Comment:
you shouldn't do this, you are deleting element that you are going to use afterwards. while nothing will crash, the same memory can be reused for another element, do a proper goto cleanup pattern